components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/DefaultKeyValidationHandler.java

```diff
@@ -170,32 +180,69 @@ public boolean validateScopes(TokenValidationContext validationContext) throws A
                 + ":" +
                 validationContext.getHttpVerb();
 
-        boolean isValid = false;
+        Set<OAuth2ScopeValidator> oAuth2ScopeValidators = OAuthServerConfiguration.getInstance()
+                .getOAuth2ScopeValidators();
+        //validate scope for filtered validators from db
+        String[] scopeValidators;
+        OAuthAppDO appInfo;
+        OAuthAppDAO oAuthAppDAO = new OAuthAppDAO();
+
         try {
-            Set<OAuth2ScopeValidator> oAuth2ScopeValidators = OAuthServerConfiguration.getInstance()
-                    .getOAuth2ScopeValidators();
-            //validate scope from all the validators
+            appInfo = oAuthAppDAO.getAppInformation(clientId);
+            scopeValidators = appInfo.getScopeValidators();     //get scope validators from the DB
+            List<String> appScopeValidators=  new ArrayList<>(Arrays.asList(scopeValidators));
+
+            if (ArrayUtils.isEmpty(scopeValidators)) {
+                log.debug(String.format("There is no scope validator registered for %s@%s", appInfo.getApplicationName(),
```

Comment:
Isn't it better to print `scopeValidators` outside the if block? This will help us to figure out the scope validators available even if the list is empty.

Reply:
Here it is printed the message as
"There is no scope validator registered for %s@%s", appInfo.getApplicationName(), OAuth2Util.getTenantDomainOfOauthApp(appInfo)" inside the if block. Not the scopeValidators availble currently.